docs/interaction/rpc.md

```diff
@@ -6,122 +6,675 @@ sidebar_label: RPC
+      }
+    ]
+  }
+}
 ```
 
-## Query
 
-`query(path: string, data: bytes)`: queries information in the state machine / database. Where `path` can be:
+### `access_key/<account_id>/<public_key>`
+
+Returns details about access key for given account with this public key. If there is no such access key, returns nothing.
 
-* `account/<account_id>` - returns view of account information, e.g. `{"amount": 1000000, "nonce": 102, "account_id": "test.near"}`
-* `access_key/<account_id>` - returns all access keys for given account.
-* `access_key/<account_id>/<public_key>` - returns details about access key for given account with this public key. If there is no such access key, returns nothing.
-* `contract/<account_id>` - returns full state of the contract \(might be expensive if contract has large state\).
-* `call/<account_id>/<method name>` - calls `<method name>` in contract `<account_id>` as view function with `data` as parameters.
 
-`http post http://127.0.0.1:3030/ jsonrpc=2.0 method=query params:="[\"account/test.near\",\"\"]" id="dontcare"`
+**`FullAccess` key**
+
+```bash
+http post https://rpc.nearprotocol.com jsonrpc=2.0 method=query params:='["access_key/near.test/ed25519:4Ynh6YmogjUYc5V9VqtQ1pRnZ4KuJSAJdkuj9hLj4a2p", ""]' id=dontcare
+```
 
-```javascript
+```json
 {
-    "id": "dontcare",
-    "jsonrpc": "2.0",
-    "result": {
-        "code": 0,
-        "codespace": "",
-        "height": 0,
-        "index": -1,
-        "info": "",
-        "key": "YWNjb3VudC90ZXN0Lm5lYXI=",
-        "log": "exists",
-        "proof": [],
-        "value": "eyJhY2NvdW50X2lkIjoidGVzdC5uZWFyIiwibm9uY2UiOjAsImFtb3VudCI6MTAwMDAwMDAwMDAwMCwic3Rha2UiOjUwMDAwMDAwLCJwdWJsaWNfa2V5cyI6W1sxNjIsMTIyLDE0MCwyMTksMTcyLDEwNSw4MCw3OCwxOTAsMTY1LDI1NSwxNDAsMTExLDQzLDIyLDE0OSwyMTEsMTUyLDIyNywyMjcsNjcsMjIyLDIzNCw3Nyw5NiwxNTYsNjYsMjMsMTcyLDk2LDc2LDEzN11dLCJjb2RlX2hhc2giOiJBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBPSJ9"
+  "id": "dontcare",
+  "jsonrpc": "2.0",
+  "result": {
+    "block_height": 288274,
+    "nonce": 1,
+    "permission": "FullAccess"
+  }
+}
+```
+
+**`FunctionCall` access key**
+
+```bash
+http post https://rpc.nearprotocol.com jsonrpc=2.0 method=query params:='["access_key/near.test/ed25519:9UfaMwBArFmccTYtzq43K3Kdoncp4kd8rx7cWk57oasL", ""]' id=dontcare
+```
+
+```json
+{
+  "id": "dontcare",
+  "jsonrpc": "2.0",
+  "result": {
+    "block_height": 288293,
+    "nonce": 0,
+    "permission": {
+      "FunctionCall": {
+        "allowance": "100000000",
+        "method_names": [],
+        "receiver_id": "studio-d4zxn4xdu"
+      }
     }
+  }
 }
 ```
 
-Where `value` is base58 encoded JSON of the account status: `'{"account_id":"test.near","nonce":0,"amount":1000000000000,"stake":50000000,"public_keys":[[...]],"code_hash":"AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA="}'`
+### `contract/<account_id>`
+
+Returns full state of the contract (might be expensive if contract has large state).
```

Comment:
Heads up!

This method is currently in a broken state. @ilblackdragon has a PR to fix this https://github.com/nearprotocol/nearcore/pull/2075, and the API is:

```
$ http post http://localhost:3030 jsonrpc=2.0 method=query id=123 params:='["contract/my_token", ""]'
```
```js
{
    "id": "123",
    "jsonrpc": "2.0",
    "result": {
        "block_height": 2057,
        "values": [
            {
                "key": "U1RBVEU=",
                "value": "AgAAAGFpAQAAAAAAAAACAAAAYWsBAAAAAAAAAAIAAABhdqCGAQAAAAAAAAAAAAAAAAA="
            },
            {
                "key": "YWkMAAAAbXlfdmFsaWRhdG9y",
                "value": "AAAAAAAAAAA="
            },
            {
                "key": "YWsAAAAAAAAAAA==",
                "value": "DAAAAG15X3ZhbGlkYXRvcg=="
            },
            {
                "key": "YXYAAAAAAAAAAA==",
                "value": "oIYBAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
            }
        ]
    }
}
```

The second input parameter is base58-encoded state key prefix. It is worth noting that the new API #184 uses base64-encoded values. The output keys and values are encoded with base64 (the output is the same for current and the new APIs).

Reply:
✅